### PR TITLE
Fixes #32045 - Add start-end date pickers to job wizard

### PR DESCRIPTION
**Pull request by MariaAga**
Updates vendor js to 8.4.0 in order to use datepicker and timepicker.
Depends on https://github.com/theforeman/foreman_remote_execution/pull/558

### CODE REVIEW

**Comment by MariaAga** [comment]
js tests are failing because the tests here are using a newer version of pf where they use `time` instead of `defaulttime`

File: webpack/JobWizard/steps/Schedule/__tests__/Schedule.test.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+const lodash = require('lodash');
+
+lodash.debounce = fn => fn;
+jest.spyOn(api, 'get');
+jest.spyOn(selectors, 'selectJobTemplate');
+jest.spyOn(selectors, 'selectJobTemplates');
+jest.spyOn(selectors, 'selectJobCategories');
+
+jest.spyOn(api, 'get');
```
Comment (MariaAga) [inline webpack/JobWizard/steps/Schedule/__tests__/Schedule.test.js:18]: duplicate of line 13

File: webpack/JobWizard/steps/form/DateTimePicker.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      const parsedNewTime = new Date(`${formattedDate} ${newTime}`);
+      setDateTime(parsedNewTime.toString());
+    }
+  };
+  return (
+    <>
+      <DatePicker
+        value={formattedDate}
+        placeholder="yyyy/mm/dd"
+        onChange={debounce(onDateChange, 1000, {
```
Comment (MariaAga) [inline webpack/JobWizard/steps/form/DateTimePicker.js:50]: Any opinions on the wait amount before formatting the users input?

File: webpack/JobWizard/steps/form/DateTimePicker.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  const onTimeChange = newTime => {
+    if (isValidTime(newTime)) {
+      const parsedNewTime = new Date(`${formattedDate} ${newTime}`);
+      setDateTime(parsedNewTime.toString());
+    }
+  };
+  return (
+    <>
+      <DatePicker
```
Comment (MariaAga) [inline webpack/JobWizard/steps/form/DateTimePicker.js:47]: should add translation to invalidFormatText

**Comment by MariaAga** [comment]
@laviro This is ready for a review :partying_face:

**Comment by adamruzicka** [comment]
Needs a rebase

**Comment by adamruzicka** [comment]
The date and time inputs do not seem to have the same height.

image

**Comment by MariaAga** [comment]
That might be a pf4 version issue as the time picker was made to be larger on the newer version

**Comment by adamruzicka** [comment]
The calendar "escapes" the form, which is what I'd expect to happen
image

However time select is constrainted within the form, which leads to two vertical scroll bars appearing on page. I'd say it should either expand the form vertically or even better, display over other elements on the page
image

**Comment by MariaAga** [comment]
Added `menuAppendTo` to the timepicker so the menu will be displayed over other elements

**Comment by adamruzicka** [comment]
Looks good to me. @laviro could you take a look?

File: package.json [hunk 4]
```diff
@@ -21,18 +21,18 @@
   },
   "devDependencies": {
     "@babel/core": "^7.7.0",
-    "@theforeman/builder": "^4.14.0",
-    "@theforeman/eslint-plugin-foreman": "^4.14.0",
-    "@theforeman/stories": "^4.14.0",
-    "@theforeman/test": "^4.14.0",
-    "@theforeman/vendor-dev": "^4.14.0",
+    "@theforeman/builder": "^8.7.0",
```
Comment (laviro) [inline package.json:24]: Let's split this into another PR? current version is 8.10.0

File: webpack/JobWizard/JobWizard.js [hunk 5]
```diff
@@ -16,6 +16,14 @@ export const JobWizard = () => {
...
   const [category, setCategory] = useState('');
   const [advancedValues, setAdvancedValues] = useState({});
+  const [scheduleValue, setScheduleValue] = useState({
+    repeatType: repeatTypes.noRepeat,
+    repeatAmount: '',
+    starts: '',
+    ends: '',
+    isFuture: false,
+    isNeverEnds: false,
+  });
```
Comment (laviro) [inline webpack/JobWizard/JobWizard.js:26]: how about moving it to a const? could be used later on?
e.g:
```js
const [scheduleValue, setScheduleValue] = useState(initialScheduleState)
```

File: webpack/JobWizard/steps/Schedule/__tests__/Schedule.test.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    expect(endsDateField.disabled).toBeFalsy();
+    expect(endsTimeField.disabled).toBeFalsy();
+    await act(async () => {
+      fireEvent.click(neverEnds);
+    });
+    expect(neverEnds.checked).toBeTruthy();
+    expect(endsDateField.disabled).toBeTruthy();
+    expect(endsTimeField.disabled).toBeTruthy();
+  });
+});
```
Comment (laviro) [inline webpack/JobWizard/steps/Schedule/__tests__/Schedule.test.js:155]: Great test!

**Review by laviro — COMMENTED** [review]
Thanks @MariaAga ! can you also rebase the PR?

File: package.json (hunk 4 above)
Comment (MariaAga) [reply package.json:24]: https://github.com/theforeman/foreman_remote_execution/pull/646

**Comment by MariaAga** [comment]
Rebased and will now need #646
